initial windows ESP-IDF wolfssh component setup

gojimmypi opened the pull request:
initial [setup_win.bat](https://github.com/wolfSSL/wolfssh/compare/master...gojimmypi:ESP32_component?expand=1#diff-9dda2c0b133c2b0bcb2d5b28ac8d88d97b1aac3b803ed402dba0fe73a465d9dc) for the ESP-IDF `wolfssh` and a placeholder for `setup.sh`

gojimmypi commented:
@ejohnstown  please stand by. I may need to fix up an additional item here before merge.

ejohnstown commented on ide/Espressif/ESP-IDF/setup_win.bat:
```diff
@@ -0,0 +1,415 @@
+
+echo;
+echo Copying CMakeLists.txt to %WOLFSSHLIB_TRG_DIR%\
+xcopy %WOLFSSH_ESPIDFDIR%\libs\CMakeLists.txt                %WOLFSSHLIB_TRG_DIR%\                             /F
```
wolfSSH doesn't have a CMakeLists.txt file.

gojimmypi replied:
the new script now includes default make files that were missing

ejohnstown commented on ide/Espressif/ESP-IDF/setup_win.bat:
```diff
@@ -0,0 +1,415 @@
+:: SSH
+set WOLFSSH_ESPIDFDIR=%BASEDIR%\IDE\Espressif\ESP-IDF
+
+set WOLFSSHLIB_TRG_DIR=%IDF_PATH%\components\wolfssh
```
Can you quote the path? My path was "\Users\John Safranek", and was getting treated as multiple parameters to xcopy. (I did link my home directory to "\Users\John", which worked.)

gojimmypi replied:
added quotes to paths

gojimmypi commented:
Hi @ejohnstown - I've removed the generation of `options.h` in favor of `#define WOLFSSL_USER_SETTINGS` and `user_settings.h`. Please check and confirm I understood and implemented as desired.

gojimmypi commented:
fyi @ejohnstown @dgarske from [ESP-IDF Getting Started](https://docs.espressif.com/projects/esp-idf/en/v4.4.1/esp32/get-started/index.html#step-5-start-a-project):

![image](https://user-images.githubusercontent.com/13059545/170088340-dbfd4bfa-109a-48ab-aaff-bf203b1603ad.png)

Still, I'm glad the extra time was spent on supporting spaces in paths.
